pkg/util/ipset/ipset.go

```diff
@@ -421,44 +418,39 @@ func getIPSetVersionString(exec utilexec.Interface) (string, error) {
 
 // checks if port range is valid. The begin port number is not necessarily less than
 // end port number - ipset util can accept it.  It means both 1-100 and 100-1 are valid.
-func validatePortRange(portRange string) bool {
+func validatePortRange(portRange string) error {
 	strs := strings.Split(portRange, "-")
 	if len(strs) != 2 {
-		klog.Errorf("port range should be in the format of `a-b`")
-		return false
+		return fmt.Errorf("port range should be in the format of `a-b`")
 	}
 	for i := range strs {
 		num, err := strconv.Atoi(strs[i])
 		if err != nil {
-			klog.Errorf("Failed to parse %s, error: %v", strs[i], err)
-			return false
+			return fmt.Errorf("Failed to parse %s, error: %v", strs[i], err)
 		}
 		if num < 0 {
-			klog.Errorf("port number %d should be >=0", num)
-			return false
+			return fmt.Errorf("port number %d should be >=0", num)
 		}
 	}
-	return true
+	return nil
 }
 
 // checks if the given ipset type is valid.
-func validateIPSetType(set Type) bool {
+func validateIPSetType(set Type) error {
 	for _, valid := range ValidIPSetTypes {
 		if set == valid {
-			return true
+			return nil
 		}
 	}
-	klog.Errorf("Currently supported ipset types are: %v, %s is not supported", ValidIPSetTypes, set)
-	return false
+	return fmt.Errorf("Currently supported ipset types are: %v, %s is not supported", ValidIPSetTypes, set)
```

Comment:
`"unsupported SetType: %q"` - the possible types are already enumerated in types.go